Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -498,8 +498,13 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 			}
 		}
 
-		// Find the entry which hasn't been used for the longest time
-		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount)
+		// Find the entry which hasn't been used for the longest time. Some games create
+		// animations by only changing the palette. Keeping all these textures in the cache
+		// improves the performance a lot for these games, so ignore paletted textures with
+		// the same base texture here. Example: Sonic the Fighters (inside Sonic Gems
+		// Collection) loops a 64 frames animation, where only the palette changes
+		if (entry->frameCount != FRAMECOUNT_INVALID && entry->frameCount < temp_frameCount &&
+			!(isPaletteTexture && entry->base_hash == tex_hash))
```

Comment:
Yeah, if the game uses this 64 frames animation and something else at the same offset, Dolphin would overwrite parts of the animation for this something else or vice versa.

The question now is which is worse:
1. Not catching paletted texure animation and something else at the same offset
2. Adding a routine to set texture entries to used when using a paletted texture

I'd say 2) is worse